google/oauth2/service_account.py

```diff
@@ -436,6 +455,26 @@ def refresh(self, request):
                 "domain wide delegation is not supported for non-default universe domain"
             )
 
+        # If a refresh is triggered under the case of credential not expired
+        # that means some abnormal cases proactively trigger this refresh, we
+        # will attemp to fetch trust boundary but tolerant on errors if we
+        # already have cached trust boundaries.
+        if (
+            self.valid
+            and self.universe_domain in _client.TRUST_BOUNDARY_ENABLED_UNIVERSES
+            and self._trust_boundary_enabled
+        ):
+            try:
+                self._trust_boundary = self.lookup_trust_boundary(request)
+            except Exception as err:
+                if not self._trust_boundary:
+                    raise err
+                # if we already have trust boundary values, we log the error
+                # and keep the cached trust boundary value to let it fail
+                # afterwards.
+                else:
+                    _LOGGER.error(f"trust boundary refresh failed due to {err}")
```

Comment:
should this be a warning instead of an error? Also, explain that a cached value was used instead.

Reply:
+1 to warning. 
```suggestion
                    _LOGGER.warning(f"Failed to refresh trust boundary cache due to {err}. Falling back to existing cache.")
```